css-view-transitions-2/Overview.bs

```diff
@@ -0,0 +1,373 @@
+	To <dfn>reveal {{Document}}</dfn> |document|:
+		1. Assert: |document|'s [=page showing=] is false.
+
+		1. If |document| is [=render-blocked=] then return.
+
+		1. Fire a new event named <code>reveal</code> on |document|'s [=relevant global object=],
+			using {{PageRevealEvent}}.
+
+		1. Let |transition| be the result of getting the [=inbound cross-document view-transition=] for |document|.
+
+		1. If |transition| is not null, then [=activate view transition|activate=] |transition|.
+	</div>
+
+## Setting up and activating the cross-document view transition
+
+	<div algorithm>
+		To <dfn>setup outbound cross-document view transition</dfn> given a {{Document}} |oldDocument|
+		and a {{Document}} |newDocument|:
+
+		1. Assert: this is called when |oldDocument| is [=unloading=].
+
+		1. If |oldDocument|'s [=active view transition=] is not null,
+			then [=skip the view transition|skip=] |oldDocument|'s [=active view transition=]
+			with an "{{AbortError}}" {{DOMException}} in |oldDocument|'s [=relevant Realm=].
+
+		1. If |oldDocument|'s [=environment settings object/origin=] is not [=same origin=] as
+			|newDocument|'s [=environment settings object/origin=] then return.
+
+		1. If |newDocument| [=was created via cross-origin redirects=] is true
+			and |newDocument|'s [=latest entry=] is null, then return.
+
+			Note: A document with a non-null [=latest entry=]
+			is being [=Document/reactivated=], in which case we don't need to check for cross-origin redirects.
+
+		1. If |oldDocument| does not [=opt in to cross-document view transitions=], then return.
+
+			Note: we don't know yet if |newDocument| has opted in, as it might not be parsed yet.
+			We check the opt-in for |newDocument| when it is [=reveal document|revealed=].
+
+		1. Let |transition| be a new {{ViewTransition}} object in |oldDocument|'s [=relevant Realm=].
+
+		1. [=Capture the old state=] for |transition|. If this fails, return.
+
+			Note: implementations might need to [=queue a task=] in order to [=capture the old state=].
+			However, this should not have web-observable side-effects.
+
+		1. [=Queue a global task=] on the [=DOM manipulation task source=],
+			given |newDocument|'s [=relevant global object=], to perform the following step:
+
+			1. Set |newDocument|'s [=active view transition=] to a new {{ViewTransition}} in |newDocument|'s [=relevant Realm=],
+				whose [=ViewTransition/named elements=] is |transition|'s [=ViewTransition/named elements=],
+				[=ViewTransition/update callback done promise=] is [=a promise resolved with=] undefined,
+				[=ViewTransition/initial snapshot containing block size=] is |transition|'s [=ViewTransition/initial snapshot containing block size=],
+				and whose [=ViewTransition/is cross-document=] is true.
+
+			Note: this task would run before any script on the new {{Document}}.
+	</div>
+
+	<div algorithm>
+		To get the <dfn>inbound cross-document view-transition</dfn> for a {{Document}} |document|:
```

Comment:
Why do we need a new variable? Shouldn't this just use "active view transition". This raises the question of what happens if author calls startVT before the cross-doc transition is done. I feel like we should give the same-doc VT priority in the new Doc but this deserves an issue.

Overall this means we can't support concurrent transitions which are scoped to the same element. @vmpstr and I discussed this and align on that restriction. MPA and SPA transitions are scoped to the root element today. This restriction will get relaxed with scoped transitions, where an author could do a same-doc transition scoped to body for example.

Reply:
I mistook "inbound cross-document view-transition" to be a variable instead of algorithm. My bad, this LGTM.